### DISCUSSION TITLE
Modify the route itself from within an onRoute hook

### DISCUSSION THREAD

**Discussion by my-lalex** [fastify/fastify#5036] [Q&A]
I'm trying to create a automatic route-parameter typing plugin. The point would be to handle all the routes with a onRoute hook, and modify the path and schema, knowing that I have to handle prefixes to...

```javascript
{
	path: '/company/:companyId(number)/users'
}

// would mutate route properties to

{
	path: '/company/:companyId/users',
	schema: {
		params: {
			type:"object",
			required:['companyId'],
			properties: {
				companyId: { type:'number' }
			}
		}
	}
}
```

Is there a way to do this?

IMHO the `path`/`url` alias  with a `routePath` field in the hook is very confusing: when prefixed, we don't get in the `onRoute` hook the same value we used for route registering...
I would have use only the `path` field for route creation, and give the `url` as read-only info. 

BTW, I get that this modification would be a huge breaking change: easy to implement, but not so easy to put in place.

**Comment by metcoder95** [comment]
The plugins sound interesting 👍 

Moving to your problem, I believe it should be possible. In theory `path/url` is the same as pointing to the whole route path as defined by the user + prefix, `routePath` is the path which the route is being registered under.

What's the result your are currently having, do you have an 
[Minimum Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) you can share?
